pkgs/by-name/sh/shadowsocks-qtun-plugin/package.nix

```diff
@@ -0,0 +1,32 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "shadowsocks-qtun-plugin";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "shadowsocks";
+    repo = "qtun";
+    rev = "v${version}";
+    hash = "sha256-JYwbDlza/EuWeEiO8KS96y7koIbpOocBKS5VZjyzP6g=";
+  };
+
+  cargoPatches = [
+    # repo does not contain lock file
+    ./cargo-lock.diff
```

Comment:
just vendor the lock file directly, like in `pkgs/by-name/te/termbg/package.nix`